courseware/api.py

```diff
@@ -306,7 +317,41 @@ def enroll_in_edx_course_runs(user, course_runs):
         requests.exceptions.HTTPError: Raised if the underlying HTTP request fails
     """
     edx_client = get_edx_api_client(user)
-    return [
-        edx_client.enrollments.create_audit_student_enrollment(course_run.courseware_id)
-        for course_run in course_runs
-    ]
+    results = []
+    for course_run in course_runs:
+        try:
+            result = edx_client.enrollments.create_student_enrollment(
+                course_run.courseware_id, mode=EDX_ENROLLMENT_PRO_MODE
+            )
+            results.append(result)
+        except HTTPError as exc:
+            # If the error message indicates that the preferred enrollment mode was the cause of the
+            # error, log an error and try to enroll the user in 'audit' mode as a failover.
+            enroll_mode_error_texts = (
```

Comment:
Nit: this can be a constant defined outside the method to reduce the method size.